docs/authentisieren.adoc

```diff
@@ -110,7 +110,7 @@ Die enthaltene Nonce gegen den im Schritt 1 erzeugten Wert abgeglichen. Anschlie
+2. Die vom IDP in der obigen Response übermittelte Challenge beinhaltet ein Challenge-Token `{"iss":"https://idp.zentral.idp.splitdns.ti-dienste.de","response_type":"code","snc":"syQAvJmxPnRtLjT6uPVERb_RF7MmVzhS1sP8FbHjhLM","code_challenge_method":"S256","token_type":"challenge","nonce":"887766","client_id":"gematikTestPs","scope":"openid e-rezept","state":"xxxstatexxx","redirect_uri":"http://test-ps.gematik.de/erezept","exp":1616686048,"iat":1616685868,"code_challenge":"Ca3Ve8jSsBQOBFVqQvLs1E-dGV1BXg2FTvrd-Tg19Vg","jti":"5e5ad23ae3e7d8aa"}` in Base64-Codierung `eyJpc3MiOiJodHRwczovL2lkcC56ZW50cmFsLmlkcC5zcGxpdGRucy50aS1kaWVuc3RlLmRlIiwicmVzcG9uc2VfdHlwZSI6ImNvZGUiLCJzbmMiOiJzeVFBdkpteFBuUnRMalQ2dVBWRVJiX1JGN01tVnpoUzFzUDhGYkhqaExNIiwiY29kZV9jaGFsbGVuZ2VfbWV0aG9kIjoiUzI1NiIsInRva2VuX3R5cGUiOiJjaGFsbGVuZ2UiLCJub25jZSI6Ijg4Nzc2NiIsImNsaWVudF9pZCI6ImdlbWF0aWtUZXN0UHMiLCJzY29wZSI6Im9wZW5pZCBlLXJlemVwdCIsInN0YXRlIjoieHh4c3RhdGV4eHgiLCJyZWRpcmVjdF91cmkiOiJodHRwOi8vdGVzdC1wcy5nZW1hdGlrLmRlL2VyZXplcHQiLCJleHAiOjE2MTY2ODYwNDgsImlhdCI6MTYxNjY4NTg2OCwiY29kZV9jaGFsbGVuZ2UiOiJDYTNWZThqU3NCUU9CRlZxUXZMczFFLWRHVjFCWGcyRlR2cmQtVGcxOVZnIiwianRpIjoiNWU1YWQyM2FlM2U3ZDhhYSJ9`. Da die Signatur immer über einen Hashwert der zu signierenden Daten erfolgt, muss dieser Hashwert vom Clientsystem berechnet werden. Als kryptografisches Verfahren kommt hier SHA-256 zum Einsatz. Aus dem obigen Beispiel `eyJpc3MiOiJodHRwczovL2lkcC56ZW50cmFsLmlkcC5...` ergibt sich folgender Hashwert: `a52c5c801f8697c6ffe1c79735cf635cf1fadcd1dfe7b09d21eb84c636564f11`
```

Comment:
Hello @christiankochsbe,
thank you for raising the issue and creating a PR!

We evaluated your comments and come to this conclusion:

So, the SHA-256 hash value was definitely wrong. However, the result `a52c5c801f8697c6ffe1c79735cf635cf1fadcd1dfe7b09d21eb84c636564f11` is not a HEX representation, but just a UTF-8 encoded character string of the JWS payload. This must be combined with the Base64 encoded header {"alg": "ES256","typ": "JWT"}:
eyJhbGciOiAiRVMyNTYiLCJ0eXAiOiAiSldUIn0.eyJpc3MiOiJodHRwczovL2lkcC56ZW50cmFsLmlkcC5zcGxpdGRucy50aS1kaWVuc3RlLmRlIiwicmVzcG9uc2VfdHlwZSI6ImNvZGUiLCJzbmMiOiJzeVFBdkpteFBuUnRMalQ2dVBWRVJiX1JGN01tVnpoUzFzUDhGYkhqaExNIiwiY29kZV9jaGFsbGVuZ2VfbWV0aG9kIjoiUzI1NiIsInRva2VuX3R5cGUiOiJjaGFsbGVuZ2UiLCJub25jZSI6Ijg4Nzc2NiIsImNsaWVudF9pZCI6ImdlbWF0aWtUZXN0UHMiLCJzY29wZSI6Im9wZW5pZCBlLXJlemVwdCIsInN0YXRlIjoieHh4c3RhdGV4eHgiLCJyZWRpcmVjdF91cmkiOiJodHRwOi8vdGVzdC1wcy5nZW1hdGlrLmRlL2VyZXplcHQiLCJleHAiOjE2MTY2ODYwNDgsImlhdCI6MTYxNjY4NTg2OCwiY29kZV9jaGFsbGVuZ2UiOiJDYTNWZThqU3NCUU9CRlZxUXZMczFFLWRHVjFCWGcyRlR2cmQtVGcxOVZnIiwianRpIjoiNWU1YWQyM2FlM2U3ZDhhYSJ9

and then be hashed: SHA256(x) = `94238882b24aaade41950ecee5a8ab14c4196ed4c5d9d2dfa344fdfd63a27262`

Can you confirm that?

Reply:
Looks good to me! 
To make this clear, the Base64 encoded payload was converted into a HEX representation which was then hashed with SHA256.

Reply:
No, not quite. The header was included in the base64 string. See this [link on CyberChef](https://gchq.github.io/CyberChef/#recipe=SHA2('256',64,160)&input=ZXlKaGJHY2lPaUFpUlZNeU5UWWlMQ0owZVhBaU9pQWlTbGRVSW4wLmV5SnBjM01pT2lKb2RIUndjem92TDJsa2NDNTZaVzUwY21Gc0xtbGtjQzV6Y0d4cGRHUnVjeTUwYVMxa2FXVnVjM1JsTG1SbElpd2ljbVZ6Y0c5dWMyVmZkSGx3WlNJNkltTnZaR1VpTENKemJtTWlPaUp6ZVZGQmRrcHRlRkJ1VW5STWFsUTJkVkJXUlZKaVgxSkdOMDF0Vm5wb1V6RnpVRGhHWWtocWFFeE5JaXdpWTI5a1pWOWphR0ZzYkdWdVoyVmZiV1YwYUc5a0lqb2lVekkxTmlJc0luUnZhMlZ1WDNSNWNHVWlPaUpqYUdGc2JHVnVaMlVpTENKdWIyNWpaU0k2SWpnNE56YzJOaUlzSW1Oc2FXVnVkRjlwWkNJNkltZGxiV0YwYVd0VVpYTjBVSE1pTENKelkyOXdaU0k2SW05d1pXNXBaQ0JsTFhKbGVtVndkQ0lzSW5OMFlYUmxJam9pZUhoNGMzUmhkR1Y0ZUhnaUxDSnlaV1JwY21WamRGOTFjbWtpT2lKb2RIUndPaTh2ZEdWemRDMXdjeTVuWlcxaGRHbHJMbVJsTDJWeVpYcGxjSFFpTENKbGVIQWlPakUyTVRZMk9EWXdORGdzSW1saGRDSTZNVFl4TmpZNE5UZzJPQ3dpWTI5a1pWOWphR0ZzYkdWdVoyVWlPaUpEWVROV1pUaHFVM05DVVU5Q1JsWnhVWFpNY3pGRkxXUkhWakZDV0djeVJsUjJjbVF0VkdjeE9WWm5JaXdpYW5ScElqb2lOV1UxWVdReU0yRmxNMlUzWkRoaFlTSjk) to see what happened.

Reply:
Ok, I see. Perhaps this should be described more clearly in the documentation. But I am fine with that. Thanks!